homeassistant/components/camera/__init__.py

```diff
@@ -0,0 +1,273 @@
+                handler.request.sendall(bytes(
+                    'Content-type: multipart/x-mixed-replace; \
+                        boundary=--jpgboundary\r\n\r\n', 'utf-8'))
+
+                handler.request.sendall(bytes('--jpgboundary\r\n', 'utf-8'))
+
+                while True:
+
+                    if camera.username and camera.password:
+                        response = requests.get(
+                            camera.still_image_url,
+                            auth=HTTPBasicAuth(
+                                camera.username,
+                                camera.password))
+                    else:
+                        response = requests.get(camera.still_image_url)
+
+                    headers_str = '\r\n'.join((
+                        'Content-length: {}'.format(len(response.content)),
+                        'Content-type: image/jpeg',
+                    )) + '\r\n\r\n'
+
+                    handler.request.sendall(
+                        bytes(headers_str, 'utf-8') +
+                        response.content +
+                        bytes('\r\n', 'utf-8'))
+
+                    handler.request.sendall(
+                        bytes('--jpgboundary\r\n', 'utf-8'))
+
+            except (requests.RequestException, IOError):
+                camera.is_streaming = False
+                camera.update_ha_state()
+
+        else:
+            handler.send_response(HTTP_NOT_FOUND)
+
+        camera.is_streaming = False
+
+    hass.http.register_path(
+        'GET',
+        re.compile(
+            r'/api/camera_proxy_stream/(?P<entity_id>[a-zA-Z\._0-9]+)'),
+        _proxy_camera_mjpeg_stream,
+        require_auth=True)
+
+
+class Camera(Entity):
+    """ The base class for camera components """
+
+    @property
+    # pylint: disable=no-self-use
+    def is_recording(self):
+        """ Returns true if the device is recording """
+        return False
+
+    @property
+    # pylint: disable=no-self-use
+    def is_streaming(self):
+        """ Returns true if the device is streaming """
+        return False
+
+    @is_streaming.setter
+    def is_streaming(self, value):
+        """ Set this to true when streaming begins """
+        pass
+
+    @property
+    # pylint: disable=no-self-use
+    def brand(self):
+        """ Should return a string of the camera brand """
+        return None
+
+    @property
+    # pylint: disable=no-self-use
+    def model(self):
+        """ Returns string of camera model """
+        return None
+
+    @property
+    # pylint: disable=no-self-use
+    def base_url(self):
+        """ Return the configured base URL for the camera """
+        return None
+
+    @property
+    # pylint: disable=no-self-use
+    def image_url(self):
+        """ Return the still image segment of the URL """
+        return None
+
+    @property
+    # pylint: disable=no-self-use
+    def device_info(self):
```

Comment:
All keys that can be in the device_info hash can be converted to properties on the Camera entity (like brand, model etc).